src/app/analytics/analytics.factory.js

```diff
@@ -35,190 +71,183 @@ const analyticsFactory = /* @ngInject */ function ($window, $timeout, envService
-                  id: productData.designationNumber,
-                  price: itemConfig.amount.toString(),
-                  brand: productData.orgId,
-                  category: productData.designationType.toLowerCase(),
-                  variant: frequencyObj.display.toLowerCase(),
-                  quantity: '1'
-                }]
+          $window.digitalData = {
+            page: {
+              category: {
+                subcategory1: siteSubSection
               }
             }
-          })
+          }
         }
-      } catch (e) {
-        // Error caught in analyticsFactory.cartAdd
+        
       }
-    },
-    cartRemove: function (item) {
-      try {
-        if (item) {
-          $window.digitalData.cart.item = [{
-            productInfo: {
-              productID: item.designationNumber,
-              designationType: item.designationType
-            },
-            price: {
-              basePrice: item.amount
-            },
-            attributes: {
-              donationType: item.frequency.toLowerCase() === 'single' ? 'one-time donation' : 'recurring donation',
-              donationFrequency: item.frequency.toLowerCase(),
-              siebel: {
-                productType: 'designation',
-                campaignCode: item.config['campaign-code']
-              }
-            }
-          }]
-          // Send GTM Advance Ecommerce event
-          if (typeof $window.dataLayer !== 'undefined') {
-            $window.dataLayer.push({
-              event: 'remove-from-cart',
-              ecommerce: {
-                currencyCode: 'USD',
-                remove: {
-                  products: [{
-                    name: item.designationNumber,
-                    id: item.designationNumber,
-                    price: item.amount.toString(),
-                    brand: item.orgId,
-                    category: item.designationType.toLowerCase(),
-                    variant: item.frequency.toLowerCase(),
-                    quantity: '1'
-                  }]
-                }
-              }
-            })
+
+      // Set donation type
+      if (productData.frequency === 'NA') {
+        cart.item[0].attributes.donationType = 'one-time donation'
+      } else {
+        cart.item[0].attributes.donationType = 'recurring donation'
+      }
+
+      // Set donation frequency
+      const frequencyObj = find(productData.frequencies, { name: productData.frequency })
+      cart.item[0].attributes.donationFrequency = frequencyObj && frequencyObj.display.toLowerCase()
+
+      // Set data layer
+      $window.digitalData.cart = cart
+      // Send GTM Advance Ecommerce event
+      $window.dataLayer = $window.dataLayer || []
+      $window.dataLayer.push({
+        event: 'add-to-cart',
+        ecommerce: {
+          currencyCode: 'USD',
+          add: {
+            products: [generateProduct(productData, {
+              price: itemConfig.amount,
+              recurringDate: moment([
```

Comment:
By default, moment sets the time to UTC with `04:00:00` as the time, but when we format it `line 24`, it grabs the user's local UTC and formats the date. Which causes issues like you mentioned.

```
const recurringDate = moment(`2023-09-15`);
<!-- Tokyo -->
recurringDate.format('MMMM D, YYYY h:mm:ss a') = September 15, 2023 1:00:00 pm

<!-- Los_Angeles -->
recurringDate.format('MMMM D, YYYY h:mm:ss a') = September 14, 2023 9:00:00 pm

<!-- New_York  -->
recurringDate.format('MMMM D, YYYY h:mm:ss a') = September 15, 2023 12:01:21 pm

```

But if we pass in the current time, which It fixes that issue.

For example:
```
const recurringDate = moment(`2023-09-15 ${moment().format('h:mm:ss a')}`);

<!-- Tokyo -->
recurringDate.format('MMMM D, YYYY h:mm:ss a') = September 16, 2023 12:55:50 am

<!-- Los_Angeles -->
recurringDate.format('MMMM D, YYYY h:mm:ss a') = September 15, 2023 8:55:50 am

<!-- New_York -->
recurringDate.format('MMMM D, YYYY h:mm:ss a') = September 15, 2023 4:55:50 pm
```

Will doing my second example be enough to fix any issues with timezones?

Reply:
The recurring dates conceptually have no time, just dates, and to be accurate, would use EST, I think, since that is what the Siebel servers use (I think) @paulsmithccci maybe you can confirm that piece?

Reply:
Correct, @wrandall22

Reply:
Correct, I'm only formatting the recurring dates to be `MMMM D, YYYY` when passing it to DataLayer, but I added the time above to show the difference when adding time to the initial `recurringDate` variable.

The date will always be worked out in UTC since that is what is happening in this file: src/common/services/giftHelpers/giftDates.service.js

It works out the start date and month in UTC. If the user's browser is in New Zealand on September 22nd, but UTC is still on September 21, The start date will be saved everywhere as the 21st since we're formating the date to UTC.

When it comes to the analytics factory, the date is in UTC, so what we can do it format the date sent to DataLayer as UTC.